### PR TITLE
Refactor redhat_preprovision method for VM Provisioning.

### PR DESCRIPTION
**Pull request by pkomanek**
Method refactoring and test creation.

This PR is based on the issue below.
ManageIQ/manageiq#12038

@miq-bot add_label refactoring
@miq-bot assign @tinaafitz

### CODE REVIEW

File: spec/content/automate/ManageIQ/Infrastructure/VM/Provisioning/StateMachines/Methods.class/__methods__/redhat_preprovision_spec.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      current_object = Spec::Support::MiqAeMockObject.new
+      current_object.parent = root_object
+      service.object = current_object
+    end
+  end
+
+  def build_notes
+    res = "Owner: #{prov_opt[:owner_first_name]} #{prov_opt[:owner_last_name]}\n"\
+          "Email: #{prov_opt[:owner_email]}\n"\
+          "Source Template: #{template.name}"
```
Comment (lfu) [inline spec/content/automate/ManageIQ/Infrastructure/VM/Provisioning/StateMachines/Methods.class/__methods__/redhat_preprovision_spec.rb]:
```
res = %Q(
Owner: #{prov_opt[:owner_first_name]} #{prov_opt[:owner_last_name]}
Email: #{prov_opt[:owner_email]}
Source Template: #{template.name}
)
```

**Comment by lfu** [comment]
It looks good except minor change.

**Comment by lfu** [comment]
Please use ```%``` over ```%Q``` to satisfy rubocop.

**Comment by djberg96** [comment]
@tinaafitz look ok?

**Pull request merged**